Readme.md

```diff
@@ -1,21 +1,146 @@
 
 ## Setup
 
-Start by running `yarn install` in the root.
+Start by running `yarn install` in the root in your terminal.
 
 ### Dependencies
 
 Since the repo is set up as yarn workspace, all the linking is done automatically.
 
+## How the Aragon OSx protocol works
+
+To review the contracts powering the Aragon OSx protocol, feel free to head to `packages/contracts`.
+
+The Aragon OSx protocol architecture is composed of two key sections:
+
+- __Core contracts__: details the functionalities all DAOs will have. It is composed of mostly 3 sections:
+    - **DAO treasury:** the DAO’s main contract and vault.
+    - **Permission manager**: manages permissions within the DAO to execute actions.
+    - **Plugin manager**: manages installed plugins within the DAO.
+- __Framework contracts__: in charge of creating and registering each deployed DAO or plugin. It contains:
+    - **DAO and Plugin Factories**: creates DAOs or plugins.
+    - **DAO and Plugin Registries**: registers into our protocol those DAOs or plugins.
+    - **Plugin Processor:** installs and uninstalls plugins into DAOs.
+
+Additionally to those two sections, we have developed several plugins DAOs can easily install upon creation. These are:
+
+- __Token Voting plugin__: enabling voting through token majority as a governance mechanism for DAOs
+- __Multisig plugin__: enabling assets to be transacted based on approval from members.
+- __Admin plugin__: enabling one account to have god-like access to the DAO.
+- __Addresslist Voting plugin__: a set of pre-selected addresses are able to vote within a DAO.
+
+Let's dive into more detail on each of these sections.
+
+### Core Contracts
+
+The *Core Contracts* describe how every DAO generated in the Aragon OSx protocol will be set up. It is extremely lean and by design and constitutes the most critical aspects of our architecture.
+
+In a nutshell, each DAO is composed of 3 interconnecting components:
+
+1. **The DAO treasury:** The DAO contract is where the **core functionality** of the DAO lies. It is the contract in charge of:
+    - Representing the identity of the DAO (ENS name, logo, description, other metadata)
+    - Holding and managing the treasury
+    - Executing arbitrary actions to:
+        - transfer assets
+        - call its own functions
+        - call functions in external contracts
+    - Providing general technical utilities (signature validation, callback handling)
+2. **The Permission Manager:** The permission manager is an integral part of any DAO and the center of our protocol architecture. It **manages permissions for the DAO** by specifying which addresses have permission to call distinct functions on contracts associated with your DAO.
+3. **Plugin Manager**: Any custom functionality can be added or removed through plugins, allowing you to **fully customize your DAO**. Some examples of plugins DAOs could install are:
```

Comment:
Just for my clarification: the plugin manager is the PluginSetupProcessor that handles the preparation and installation of a plugin?

Reply:
FYI: the PSP would be part of the framework, not of the DAO scope. DAO's have an internal permission database, but plugins are outside such a scope.

Reply:
Reworded it! Thanksssss